File: C_Programming/Practise_Programs/PP_052.c
/*
==============================================================================
 Program     : Display Digits of a Number Using while Loop

 Description : Extracts and displays each digit of a number starting from
               the least significant digit using a while loop.

 Author      : Shubham Gadhe
==============================================================================
*/

#include<stdio.h>

int main()
{
    int iNo = 751;
    int iDigit = 0;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        printf("%d\n", iDigit);
        iNo = iNo / 10;
    }    
    
    return 0;
}

/*
==============================================================================
 Sample Output
------------------------------------------------------------------------------
 1
 5
 7
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_052.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_052
//
//  Description       : This program accepts a number from the user
//                      and displays uppercase alphabets at odd
//                      positions while displaying '*' at even
//                      positions. The alphabet sequence advances
//                      in every iteration.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 07/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_052
{
    //////////////////////////////////////////////////////////////////
    //
    //  Function Name    : Display()
    //
    //  Description      : It is used to display uppercase
    //                     alphabets at odd positions and
    //                     '*' at even positions. The
    //                     character variable is incremented
    //                     during every iteration.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 07/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public static void Display(int iNo)
    {
        int iCnt = 0;
        char ch = '\0';

        for(iCnt = 1, ch = 'A'; iCnt <= iNo; iCnt++, ch++)
        {
            if(iCnt % 2 == 0)
            {
                System.out.print("*\t");
            }
            else
            {
                System.out.print(ch + " \t");
            }
        }

        System.out.println();
    }

    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number of elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      5
//
//  Sample Output :
//      A    *    C    *    E
//
//////////////////////////////////////////////////////////////////